src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,41 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.Collections;
+import java.util.List;
+
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public List<T> getAll(Class<? extends T> type) {
+        if (type == Bulldozer.class) {
+            MachineProducer<Bulldozer> producer = new BulldozerProducer();
+            return (List<T>) producer.get();
+        } else if (type == Excavator.class) {
+            MachineProducer<Excavator> producer = new ExcavatorProducer();
+            return (List<T>) producer.get();
+        } else if (type == Truck.class) {
+            MachineProducer<Truck> producer = new TruckProducer();
+            return (List<T>) producer.get();
+        }
+        return Collections.emptyList();
```

Comment:
Keep if else constructions simple. There is no need for an else statement after return. You can simplify the code by removing the else keyword and just keeping the return statement outside the conditional blocks.